src/api/spec/controllers/webui/projects/public_key_controller_spec.rb

```diff
@@ -27,7 +27,8 @@
     context 'with a project that has no public key' do
       let(:keyinfo_response) { '<keyinfo />' }
 
-      it { expect(response.status).to eq(404) }
+      it { is_expected.to redirect_to(project_show_path(project)) }
+      it { expect(flash[:error]).not_to be_nil }
```

Comment:
@mdeniz and @DavidKang are laughing at me because of how much i dont want to test the exact value of the error string :rofl: but what about this?
```ruby
it { is_expected.to set_flash[:error].to include('not', 'public key') }
```

I don't want to end up with a code base where the tests are so specific that they don't allow refactoring and basically become a carbon copy of the code. I know this is a pretty minor point since we're just talking about the error message string but its more the general mentality of thinking that we should test every tiny detail that I don't think we should get in to the habit. Basically rspec should test behaviour not implementation.